crates/noirc_abi/src/lib.rs

```diff
@@ -65,41 +65,45 @@ pub enum Sign {
 impl AbiType {
     pub fn num_elements(&self) -> usize {
         match self {
-            AbiType::Field(_) | AbiType::Integer { .. } => 1,
-            AbiType::Array { visibility: _, length, typ: _ } => *length as usize,
+            AbiType::Field | AbiType::Integer { .. } => 1,
+            AbiType::Array { length, typ: _ } => *length as usize,
             AbiType::Struct { fields, .. } => fields.len(),
         }
     }
 
     /// Returns the number of field elements required to represent the type once encoded.
     pub fn field_count(&self) -> u32 {
         match self {
-            AbiType::Field(_) | AbiType::Integer { .. } => 1,
-            AbiType::Array { visibility: _, length, typ } => typ.field_count() * (*length as u32),
+            AbiType::Field | AbiType::Integer { .. } => 1,
+            AbiType::Array { length, typ } => typ.field_count() * (*length as u32),
             AbiType::Struct { fields, .. } => {
                 fields.iter().fold(0, |acc, (_, field_type)| acc + field_type.field_count())
             }
         }
     }
+}
+
+#[derive(Clone, Debug, Deserialize)]
+pub struct AbiParameter {
+    pub name: String,
+    pub typ: AbiType,
+    pub visibility: AbiFEType,
```

Comment:
Not a blocker for this issue; I wonder if AbiFEType is the correct name now.

Reply:
Yeah, the documentation is a bit stale as it refers to a `FieldElementType` which no longer exists. We could rename this to `AbiVisibility` as there's another `Visibility` enum in the frontend struct.